modules/alloy/src/main/ts/ephox/alloy/ui/types/TypeaheadTypes.ts

```diff
@@ -36,7 +57,13 @@ export interface TypeaheadDetail extends CommonDropdownDetail<TieredData>, Input
   markers: {
     openClass: string;
   };
+
+  // Generate fields that are created by bouldering (and don't exist in TypeaheadSpec)
   previewing: Cell<boolean>;
+  // This is required so that we can find the Typeahead from the TieredMenu. We can't rely on just
+  // looking up the Typeahead's uid from the system, because the TieredMenu and Input can be in
+  // different alloy systems / motherships.
+  lazyTypeaheadComp: Cell<Optional<AlloyComponent>>;
```

Comment:
Does this have the potential to be a memory leak? Should we use a weak map?

Reply:
- [ ] investigate memory leak concern

Reply:
Update: cleared the cell onDetached.